dist/screeps.d.ts

```diff
@@ -1017,8 +1017,7 @@ interface SignDefinition {
 }
 interface StoreDefinition {
     [resource: string]: number;
-    energy?: number;
-    power?: number;
+    energy: number;
```

Comment:
The key being in the hash and set to `undefined` or not existing in the hash at all are identical cases. `undefined` is the value returned from an indexer which is just doing property lookup and it can't find the property.

Removing `undefined` from this return type signature would indicate that no matter what you pass in, you end up with a number. That's not what happens (you get `undefined` if there is 0 of that resource in the store) and so that's represented in the type signature.